### DISCUSSION TITLE
File name rules

### DISCUSSION THREAD

**Discussion by lukasvalach** [microsoft/PubSec-Info-Assistant#643] [General]
Hi,
Is there a specific filename policy in place? I came across this topic here: [link to Microsoft's naming and referencing containers, blobs, and metadata](https://learn.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#blob-names), but it seems that in some cases, these rules may not suffice. Are you planning to implement additional functions to address this? Some of our customers have complex file naming policies, resulting in many files ending up in an error state. I've attempted to implement a short function like so:
`const checkFileName = (fileName) => {
    const reservedCharactersRegex = /[<>"\/\\|?*& ]/g;
    const validFileName = fileName.replace(reservedCharactersRegex, '_')
                                  .replace(/ /g, "_")
                                  .replace(/-/g, '_')
                                  .replace(/–/g, '_')
                                  .normalize("NFD")
                                  .replace(/[\u0300-\u036f]/g, "");
    return validFileName;
};
`
  This function is designed to handle suspicious characters found in files, which often lead to error states. However, it doesn't work in all cases. While I understand that filenames should generally adhere to certain rules, renaming hundreds of files can be quite challenging for customers.

Additionally, sometimes files get stuck in an "Uploaded" state and won't change.

If you have any advice or recommendations, please let me know. Thank you.

**Comment by lukasvalach** [comment]
If the file failed, then I can see this in the cosmos db, data explorer.
`raise error\nazure.core.exceptions.ResourceNotFoundError: Operation returned an invalid status 'The specified blob does not exist.'\nErrorCode:BlobNotFound\n"`

**Comment by ymac3** [comment]
What I've done was to get the actual blob path from file_uri for a workaround in the enrichment process. Waiting for better approach.

```
decoded_file_name = unquote(chunk_dict["file_uri"])
proper_blob_path = re.sub(r".*.blob.core.windows.net/", "", decoded_file_name)
tag_list = get_tags(proper_blob_path)
```
